WordPress/gutenberg
# Fix showing double icons for connected blocks in pattern editor

## What?

Continued from https://github.com/WordPress/gutenberg/pull/61560. Fix https://github.com/WordPress/gutenberg/pull/61560#issuecomment-2142432095.

Prevent double icons from showing when editing the synced pattern itself.

## Why?

Showing double icons is confusing to the users.

## How?

Only show the binding indicator when editing an overridable pattern as suggested in the issue.

## Testing Instructions




1. Create a synced pattern with overridable blocks (follow the testing instructions in https://github.com/WordPress/gutenberg/pull/61560)
2. Expect there to be no double icons when editing the pattern itself in the block toolbar.
3. Expect to see the binding indicator icon when editing the overridable pattern instance.

## Screenshots or screencast 
Before:
![image](https://github.com/WordPress/gutenberg/assets/7753001/ed57c2ce-5ec6-4ab6-a678-f85ae43af12e)

After:
![image](https://github.com/WordPress/gutenberg/assets/7753001/bf739e80-9767-4a7c-a876-95c157fe1aad)
